### PR TITLE
[32058] Marked classes and interfaces as API part 3

### PR DESCRIPTION
**Pull request by Usik2203**
### Description (*)

This PR fixes https://github.com/magento/magento2/issues/32058

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32058

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests EE, WebAPI Tests

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8804](https://jira.corp.magento.com/browse/ENGCOM-8804) has been created to process this Pull Request

File: app/code/Magento/Customer/Setup/CustomerSetup.php [hunk 1]
```diff
@@ -234,7 +257,7 @@ public function getDefaultEntities()
                         'label' => 'Date of Birth',
                         'input' => 'date',
                         'frontend' => \Magento\Eav\Model\Entity\Attribute\Frontend\Datetime::class,
```
Comment (hoangngo-calvin) [inline app/code/Magento/Customer/Setup/CustomerSetup.php:259]: why don't we simplify this?

File: app/code/Magento/Sales/Model/Order/Pdf/Total/DefaultTotal.php [hunk 2]
```diff
@@ -104,7 +113,7 @@ public function getFullTaxInfo()
                 $tax['font_size'] = $fontSize;
             }
         } else {
-            /** @var $orders \Magento\Tax\Model\ResourceModel\Sales\Order\Tax\Collection */
+            /** @var $orders Collection */
```
Comment (eduard13) [inline app/code/Magento/Sales/Model/Order/Pdf/Total/DefaultTotal.php:116]:
```suggestion
            /** @var Collection $orders */
```

File: app/code/Magento/Customer/Setup/CustomerSetup.php (hunk 1 above)
Comment (eduard13) [reply app/code/Magento/Customer/Setup/CustomerSetup.php:259]: @Usik2203 could you please import this one as well?
Thanks.

File: app/code/Magento/Customer/Setup/CustomerSetup.php [hunk 3]
```diff
@@ -127,11 +150,11 @@ public function getDefaultEntities()
...
             'customer' => [
-                'entity_type_id' => \Magento\Customer\Api\CustomerMetadataInterface::ATTRIBUTE_SET_ID_CUSTOMER,
-                'entity_model' => \Magento\Customer\Model\ResourceModel\Customer::class,
-                'attribute_model' => \Magento\Customer\Model\Attribute::class,
+                'entity_type_id' => CustomerMetadataInterface::ATTRIBUTE_SET_ID_CUSTOMER,
+                'entity_model' => Customer::class,
+                'attribute_model' => Attribute::class,
                 'table' => 'customer_entity',
-                'increment_model' => \Magento\Eav\Model\Entity\Increment\NumericValue::class,
+                'increment_model' => NumericValue::class,
```
Comment (eduard13) [inline app/code/Magento/Customer/Setup/CustomerSetup.php:157]: Also check below for `\Magento\Customer\Model\ResourceModel\Attribute\Collection::class`, please import it as well

File: app/code/Magento/Checkout/Controller/Cart.php [hunk 4]
```diff
@@ -96,10 +109,10 @@ protected function _isInternalUrl($url)
         /**
          * Url must start from base secure or base unsecure url
          */
-        /** @var $store \Magento\Store\Model\Store */
+        /** @var $store Store */
```
Comment (eduard13) [inline app/code/Magento/Checkout/Controller/Cart.php:112]:
```suggestion
        /** @var Store $store */
```

**Review by eduard13 — COMMENTED** [review]
@Usik2203 could you please review the below small notes?
Thank you.

File: app/code/Magento/Customer/Setup/CustomerSetup.php [hunk 5]
```diff
@@ -127,11 +150,11 @@ public function getDefaultEntities()
...
                 'additional_attribute_table' => 'customer_eav_attribute',
                 'entity_attribute_collection' => \Magento\Customer\Model\ResourceModel\Attribute\Collection::class,
```
Comment (eduard13) [inline app/code/Magento/Customer/Setup/CustomerSetup.php:159]: Please import this one as well
```suggestion
                'entity_attribute_collection' => \Magento\Customer\Model\ResourceModel\Attribute\Collection::class,
```

**Comment by gabrieldagama** [comment]
Closed in favour of https://github.com/magento/magento2/pull/32735

**Pull request closed without merging**